.github/workflows/tests.yaml

```diff
@@ -314,8 +314,9 @@ jobs:
       - name: Install Homebrew Bundler RubyGems
         if: steps.cache.outputs.cache-hit != 'true'
         run: brew install-bundler-gems
-      - name: Install Multipass
+      - name: Install dependencies with homebrew
         run: |
+          brew install libgit2@1.7  # For building pygit2
```

Comment:
I'm not sure about this fix. There are MacOS wheels for `pygit2`, for both python `3.10` / `3.12` 
https://pypi.org/project/pygit2/1.14.1/#files

Is it possible that on `MacOS`, you're building `charmcraft` with `python3.13`, which could result in this exact problem ?